server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -795,7 +795,12 @@ public final CommitStats commitStats() {
     /**
      * @return the local checkpoint for this Engine
      */
-    public abstract long getLocalCheckpoint();
+    public abstract long getProcessedLocalCheckpoint();
```

Comment:
Should we move this method to InternalEngine? It might require some castings in tests. We can do it in a follow-up if you're okay.

Reply:
++